gpytorch/functions/_inv_quad_log_det.py

```diff
@@ -67,15 +67,48 @@ def forward(
         if (probe_vectors is None or probe_vector_norms is None) and logdet:
             num_random_probes = settings.num_trace_samples.value()
             if preconditioner is None:
-                probe_vectors = torch.empty(matrix_shape[-1], num_random_probes, dtype=dtype, device=device)
-                probe_vectors.bernoulli_().mul_(2).add_(-1)
+                if settings.deterministic_probes.on():
+                    if settings.deterministic_probes.probe_vectors is None:
```

Comment:
Let's add a warning here that this will cause weird behavior when training multiple models with this setting on b/c of the global scope of the caching.